check_no_globals.go

```diff
@@ -10,8 +10,42 @@ import (
 	"strings"
 )
 
-func isWhitelisted(i *ast.Ident) bool {
-	return i.Name == "_" || i.Name == "version" || looksLikeError(i)
+func isWhitelisted(v ast.Node) bool {
+	switch i := v.(type) {
+	case *ast.Ident:
+		return i.Name == "_" || i.Name == "version" || looksLikeError(i)
+	case *ast.CallExpr:
+		if expr, ok := i.Fun.(*ast.SelectorExpr); ok {
+			return isWhitelistedSelectorExpression(expr)
+		}
+	case *ast.CompositeLit:
+		if expr, ok := i.Type.(*ast.SelectorExpr); ok {
+			return isWhitelistedSelectorExpression(expr)
+		}
+	}
+
+	return false
+}
+
+func isWhitelistedSelectorExpression(v *ast.SelectorExpr) bool {
+	x, ok := v.X.(*ast.Ident)
+	if !ok {
+		return false
+	}
+
+	allowedCombinations := [][]string{
+		{"errors", "New"},
+		{"fmt", "Errorf"},
+		{"regexp", "MustCompile"},
+	}
+
+	for _, i := range allowedCombinations {
+		if x.Name == i[0] && v.Sel.Name == i[1] {
```

Comment:
Could we make `i` a type with fields `Name` and `SelName` so the above combinations reads clearer and we about the indexes here?

Reply:
Yeah, sounds like a good idea! I'll fix that!